src/ai/backend/manager/api/utils.py

```diff
@@ -173,17 +175,24 @@ async def wrapped(request: web.Request, *args, **kwargs) -> web.Response:
                     orig_params = dict(request.query)
                 stripped_params = orig_params.copy()
                 log.debug("stripped raw params: {}", mask_sensitive_keys(stripped_params))
-                checked_params = checker.check(stripped_params)
+                if isinstance(checker, t.Trafaret):
+                    checked_params = checker.check(stripped_params)
+                else:
+                    checked_params = checker.model_validate(stripped_params)
```

Comment:
For extra type safety, let's use the pattern matching.

```suggestion
                match checker:
                    case t.Trafaret():
                        checked_params = checker.check(stripped_params)
                    case BaseModel():
                        checked_params = checker.model_validate(stripped_params)
```